localstack-core/localstack/services/stepfunctions/asl/component/program/program.py

```diff
@@ -72,7 +72,7 @@ def _get_state(self, state_name: str) -> CommonStateField:
     def eval(self, env: Environment) -> None:
         timeout = self.timeout_seconds.timeout_seconds if self.timeout_seconds else None
         env.next_state_name = self.start_at.start_at_name
-        worker_thread = threading.Thread(target=super().eval, args=(env,))
+        worker_thread = threading.Thread(target=super().eval, args=(env,), daemon=True)
```

Comment:
Yes good point, I first want to understand properly the internal guidelines on when to use or not this class. I would like future maintenance on FuncThread to be conceptually compatible with the idea of running parallel processes rather than background functions.